src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,446 @@
+
+~~~ sql
+DECLARE
+	c CURSOR FOR query;
+~~~
+
+Note that the preceding cursor still has the `REFCURSOR` data type.
+
+For information about opening and using cursors, see [Open and use cursors](#open-and-use-cursors).
+
+### Assign a result to a variable
+
+Use the PL/pgSQL `INTO` clause to assign a result of a [`SELECT`]({% link {{ page.version.version }}/select-clause.md %}) or mutation ([`INSERT`]({% link {{ page.version.version }}/insert.md %}), [`UPDATE`]({% link {{ page.version.version }}/update.md %}), [`DELETE`]({% link {{ page.version.version }}/delete.md %})) statement to a specified variable:
+
+~~~ sql
+SELECT expression INTO target FROM ...;
+~~~
+
+~~~ sql
+[ INSERT | UPDATE | DELETE ] ... RETURNING expression INTO target;
+~~~
+
+- `expression` is an [expression](https://www.postgresql.org/docs/16/plpgsql-expressions.html) that defines the result to be assigned to the variable.
+- `target` is an arbitrary variable name. This can be a list of comma-separated variables, or a single [composite variable]({% link {{ page.version.version }}/create-type.md %}#create-a-composite-data-type).
+
+For example, given a table `t` with `INT` column `col`:
+
+The following [stored procedure]({% link {{ page.version.version }}/stored-procedures.md %}) inserts a specified value `x` into the table, and the `INTO` clause assigns the [returned value]({% link {{ page.version.version }}/insert.md %}#insert-and-return-values) to `i`.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE PROCEDURE p(x INT) AS $$
+	DECLARE
+	    i INT;
+	BEGIN
+	    INSERT INTO t (col) VALUES (x) RETURNING col INTO i;
+	    RAISE NOTICE 'New Row: %', i;
+	END 
+$$ LANGUAGE PLpgSQL;
+~~~
+
+When the procedure is called, it inserts the specified integer into a new row in the table, and prints a [`NOTICE`](#report-messages-and-handle-exceptions) message that contains the inserted value:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CALL p(2);
+~~~
+
+~~~
+NOTICE: New Row: 2
+CALL
+~~~
+
+The following [user-defined function]({% link {{ page.version.version }}/user-defined-functions.md %}) uses the `max()` [built-in function]({% link {{ page.version.version }}/functions-and-operators.md %}#aggregate-functions) to find the maximum `col` value in table `t`, and assigns the result to `i`.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE FUNCTION f() RETURNS INT AS $$
+	DECLARE
+		i INT;
+	BEGIN
+		SELECT max(col) INTO i FROM t;
+		RETURN i;
+	END
+$$ LANGUAGE PLpgSQL;
+~~~
+
+When the function is invoked, it displays the maximum value that was inserted into the table:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT f();
+~~~
+
+~~~
+  f
+-----
+  2
+~~~
+
+For a more extensive example of variable assignment, see [Create a stored procedure using PL/pgSQL](#create-a-stored-procedure-using-pl-pgsql).
+
+### Write conditional statements
```

Comment:
ignorable nit: not sure we need the code blocks demonstrating the conditional statements since they look like normal IF..THEN..ELSE statements to me that the user must be familiar with. I can see why we need them for the LOOP statements, but these don't seem to have any CRDB-specific things the user needs to be aware of.

Reply:
I don't want to presume that people already know this syntax, and would prefer to be explicit. Also, most of this is no more specific to CRDB than it is to PostgreSQL, but I think it's a better experience to have more detail on our side.